lib/lightning_web/live/workflow_live/github_sync_modal.ex

```diff
@@ -0,0 +1,187 @@
+defmodule LightningWeb.WorkflowLive.GithubSyncModal do
+  @moduledoc false
+  use LightningWeb, :live_component
+
+  alias Lightning.VersionControl
+  alias Phoenix.LiveView.JS
+
+  @impl true
+  def update(%{project_repo_connection: repo_connection} = assigns, socket) do
+    {:ok,
+     socket
+     |> assign(assigns)
+     |> assign_async(:verify_connection, fn ->
+       verify_connection(repo_connection)
+     end)}
+  end
+
+  defp verify_connection(repo_connection) do
+    case VersionControl.verify_github_connection(repo_connection) do
+      :ok -> {:ok, %{verify_connection: :ok}}
+      error -> error
+    end
+  end
+
+  @impl true
+  def render(assigns) do
+    ~H"""
+    <div>
+      <.modal
+        id={@id}
+        show={true}
+        close_on_keydown={false}
+        close_on_click_away={false}
+        width="min-w-1/3"
+      >
+        <:title>
+          <div class="flex justify-between">
+            <span class="font-bold">
+              Sync changes to Github
+            </span>
+
+            <button
+              phx-click="toggle_github_sync_modal"
+              type="button"
+              class="rounded-md bg-white text-gray-400 hover:text-gray-500 focus:outline-none"
+              aria-label={gettext("close")}
+            >
+              <span class="sr-only">Close</span>
+              <.icon name="hero-x-mark" class="h-5 w-5 stroke-current" />
+            </button>
+          </div>
+        </:title>
+        <div id="verify-github-connection-banner" class="mb-2">
+          <.async_result assign={@verify_connection}>
+            <:loading>
+              <div class="rounded-md bg-blue-50 p-4">
+                <div class="flex">
+                  <div class="flex-shrink-0">
+                    <svg
+                      class="animate-spin h-5 w-5 text-blue-400"
+                      xmlns="http://www.w3.org/2000/svg"
+                      fill="none"
+                      viewBox="0 0 24 24"
+                    >
+                      <circle
+                        class="opacity-25"
+                        cx="12"
+                        cy="12"
+                        r="10"
+                        stroke="currentColor"
+                        stroke-width="4"
+                      >
+                      </circle>
+                      <path
+                        class="opacity-75"
+                        fill="currentColor"
+                        d="M4 12a8 8 0 018-8V0C5.373 0 0 5.373 0 12h4zm2 5.291A7.962 7.962 0 014 12H0c0 3.042 1.135 5.824 3 7.938l3-2.647z"
+                      >
+                      </path>
+                    </svg>
```

Comment:
Could we use the `<.icon ...>` component instead ?

Reply:
I think we have loader components too with pulse animation

Reply:
Hmm, I used a raw svg because I couldn't find the circle type component in heroicons

Reply:
It would be nice to have a component for it to make the heex easier to read and to reuse as well (there are other 2 places with similar or same svg of a spinning circle).

Reply:
I've used the `arrow-path` icon instead